e2e/loadbalancer_test.go

```diff
@@ -2,36 +2,43 @@ package test
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"io/ioutil"
-	"net/http"
 	"testing"
+	"time"
 
 	"github.com/civo/civogo"
 	. "github.com/onsi/gomega"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
 func TestLoadbalancerBasic(t *testing.T) {
-
+	ctx := t.Context()
 	g := NewGomegaWithT(t)
 
-	mirrorDeploy, err := deployMirrorPods(e2eTest.tenantClient)
+	mirrorDeploy, err := deployMirrorPods(ctx, e2eTest.tenantClient)
 	g.Expect(err).ShouldNot(HaveOccurred())
-	defer e2eTest.tenantClient.Delete(context.TODO(), mirrorDeploy)
+	t.Cleanup(func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+		e2eTest.tenantClient.Delete(ctx, mirrorDeploy)
+	})
 
 	lbls := map[string]string{"app": "mirror-pod"}
 	// Create a service of type: LoadBalancer
 	svc := &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "echo-pods",
 			Namespace: "default",
+			Annotations: map[string]string{
+				"kubernetes.civo.com/firewall-id": e2eTest.cluster.FirewallID,
```

Comment:
>[!NOTE]
[After correcting the previous comment](https://github.com/civo/civo-cloud-controller-manager/pull/37/files#r2031173880), the following error occurred during service creation, so I made adjustments based on the details of that error.

```
"Event occurred" object="default/echo-pods" fieldPath="" kind="Service" apiVersion="v1" type="Warning" reason="SyncLoadBalancerFailed" message=<
        Error syncing load balancer: failed to ensure load balancer: Error: Unknown error response - status: 400 Bad Request, code: 400, reason: {"code":"firewall_missing","reason":"Please provide either instance_firewall or firewall_rule"}
```